src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java

```diff
@@ -79,14 +81,22 @@ public InternalDeliveryResult handleEmailRequest(final EmailRequest request) {
         return publishMessageEvent(message);
     }
 
-    public InternalDeliveryResult handleSmsRequest(final SmsRequest request) {
-        // Check blacklist
-        blacklistService.check(request);
+	public InternalDeliveryBatchResult handleEmailBatchRequest(final EmailBatchRequest request) {
+		var batchId = UUID.randomUUID().toString();
 
-        var message = dbIntegration.saveMessage(messageMapper.toMessage(request));
+		var deliveryResults = ofNullable(request.parties()).orElse(Collections.emptyList()).stream()
+			.filter(party -> isWhitelisted(MessageType.EMAIL, party.emailAddress()))
+			.map(party -> requestMapper.toEmailBatchRequest(request, party))
+			.map(emailRequest -> messageMapper.toMessage(emailRequest, batchId))
+			.map(dbIntegration::saveMessage)
+			.map(this::publishMessageEvent)
+			.toList();
```

Comment:
Could possibly map directly from party -> message.

Reply:
If it doesn't upset you I'l leave it "as is" for clarity :)